File: PHP/factorial.php
<?php
/*
=====================================================
PRACTICAL: PHP Program to Find Factorial of a Number
=====================================================

AIM:
- To write a PHP program that calculates the factorial
  of a given number.

OBJECTIVE:
- To read a number from user.
- To calculate factorial using loop.
- To display result.

TOOLS:
- PHP 7+/8+
- Any Web Server (XAMPP/WAMP/LAMP)

ALGORITHM:
1) Accept a number from user (through form).
2) Initialize factorial = 1.
3) Run loop i = 1 to number:
   factorial = factorial * i.
4) Print factorial result.
5) End.

STEPS TO RUN:
- Save file as: factorial.php
- Open in browser: http://localhost/factorial.php
- Enter number and check.

=====================================================
PHP CODE (factorial.php)
=====================================================
*/
?>

<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>Factorial Program</title>
</head>
<body>
  <h1>Factorial Calculation (PHP)</h1>

  <form method="post">
    <label>Enter a number:</label><br>
    <input type="number" name="num" min="0" required>
    <button type="submit">Calculate</button>
  </form>

<?php
if ($_SERVER["REQUEST_METHOD"] === "POST") {
    $num = (int)$_POST["num"];
    $fact = 1;

    if ($num < 0) {
        echo "<p>Factorial is not defined for negative numbers.</p>";
    } else {
        for ($i = 1; $i <= $num; $i++) {
            $fact *= $i;
        }
        echo "<p>Factorial of $num is $fact.</p>";
    }
}
?>

<?php
/*
================= SAMPLE OUTPUT =================

Input: 5  
Output: Factorial of 5 is 120.

Input: 0  
Output: Factorial of 0 is 1.

=================================================
RESULT:
- Successfully implemented a PHP program that
  calculates the factorial of a given number.
=================================================
*/
?>
</body>
</html>

File: Factorial.java
/*
=====================================================
PRACTICAL: Factorial of a Number in Java
=====================================================

AIM:
- To write a Java program that finds factorial of a number.

OBJECTIVE:
- To use loop for calculating factorial.
- To take user input and compute factorial.
- To display the result.

TOOLS:
- JDK 8+ (javac, java)
- Any IDE or text editor

ALGORITHM:
1) Import Scanner class.
2) Read an integer number n from user.
3) Initialize fact = 1.
4) Loop from i = 1 to n:
      fact = fact * i
5) Print factorial value.
6) End program.

STEPS TO RUN:
- Save as: Factorial.java
- Compile: javac Factorial.java
- Run:     java Factorial

=====================================================
*/

import java.util.Scanner;

public class Factorial {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int n = sc.nextInt();

        long fact = 1;
        for (int i = 1; i <= n; i++) {
            fact *= i;
        }

        System.out.println("Factorial of " + n + " = " + fact);

        sc.close();
    }
}

/*
================= SAMPLE OUTPUT =================

Enter a number: 5
Factorial of 5 = 120

Enter a number: 7
Factorial of 7 = 5040

=================================================

RESULT:
- Successfully implemented factorial program in Java.
- Program reads an integer and calculates its factorial using loop.
=================================================
*/
